service/controller/appvalue/cache.go

```diff
@@ -0,0 +1,215 @@
+package appvalue
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/giantswarm/apiextensions/v2/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/apiextensions/v2/pkg/label"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+
+	applabel "github.com/giantswarm/app-operator/v2/pkg/label"
+	"github.com/giantswarm/app-operator/v2/service/controller/app/key"
+)
+
+func (c *AppValue) buildCache(ctx context.Context) error {
+	for {
+		lo := metav1.ListOptions{
+			LabelSelector: c.selector.String(),
+		}
+
+		res, err := c.k8sClient.G8sClient().ApplicationV1alpha1().Apps("").Watch(ctx, lo)
+		if err != nil {
+			panic(err)
+		}
+
+		for r := range res.ResultChan() {
+			cr, err := key.ToCustomResource(r.Object)
+			if err != nil {
+				panic(err)
+			}
+
+			err = c.addCache(ctx, cr, r.Type)
```

Comment:
Here we collect all configmap objects belong to an app CR, and later use it on `watch.go`